### ISSUE TITLE
Oak Functions micro RPC definition uses bytes instead of protos

### ISSUE THREAD

**Issue by tiziano88** [project-oak/oak#4267]
https://github.com/project-oak/oak/blob/60a766693354595b9700b4bf8dfdd77e3c736419/oak_functions_service/proto/oak_functions.proto#L64-L67

(I think the referenced issue is incorrect / obsolete)

@ipetr0v is this still something we should fix? If so, which exact protos should we use? Could you take a look?

**Comment by ipetr0v** [comment]
Can `micro_rpc` depend on the Protobuf generated files?
The same way it's done in https://github.com/project-oak/oak/pull/4211

**Comment by tiziano88** [comment]
I think so, could you give it a try? Do we expect to include the oak_crypto messages inside the existing messages, or to replace the messages entirely?

**Comment by ipetr0v** [comment]
I'll take a look.

> Do we expect to include the oak_crypto messages inside the existing messages, or to replace the messages entirely?

The current plan is to use explicit messages.

**Comment by ipetr0v** [comment]
Fixing this is a breaking change and requires implementing: https://github.com/project-oak/oak/issues/4037

Will start updating all the protos.

**Comment by tiziano88** [comment]
> The current plan is to use explicit messages.

My question is whether the explicit messages replace the current request / response messages, or are used as fields inside them.

FYI if using them as fields, I think changing type from `bytes` to the actual message is a non-breaking change (bytes and sub messages are encoded identically): https://protobuf.dev/programming-guides/encoding/#embedded . 

But if replacing the message entirely, then of course that's not useful.

BTW it would still be useful to think through how to make this a non-breaking change if we had to. Could you summarize that here? Then we can decide whether to do it or not.
